financial-connections/src/main/java/com/stripe/android/financialconnections/features/success/SuccessContent.kt

```diff
@@ -0,0 +1,223 @@
+            showSpinner = false
+        }
+    }
+
+    FinancialConnectionsScaffold(
+        topBar = {
+            FinancialConnectionsTopAppBar(
+                showBack = false,
+                onCloseClick = onCloseClick,
+                elevation = scrollState.elevation
+            )
+        }
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(24.dp)
+        ) {
+            Box(
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth(),
+                contentAlignment = Alignment.Center
+            ) {
+                SpinnerToSuccessAnimation(
+                    customSuccessMessage = payload.customSuccessMessage,
+                    accountsCount = payload.accountsCount,
+                    showSpinner = showSpinner
+                )
+            }
+            SuccessFooter(
+                modifier = Modifier.alpha(if (showSpinner) 0f else 1f),
+                merchantName = payload.businessName,
+                loading = completeSessionAsync is Loading,
+                onDoneClick = onDoneClick
+            )
+        }
+    }
+}
+
+@Composable
+private fun SpinnerToSuccessAnimation(
+    showSpinner: Boolean,
+    accountsCount: Int,
+    customSuccessMessage: TextResource?
+) {
+    Box(
+        modifier = Modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        // Define the animation specs
+        val enterTransition = fadeIn(animationSpec = tween(ENTER_TRANSITION_DURATION_MS))
+        val exitTransition = fadeOut(animationSpec = tween(ENTER_TRANSITION_DURATION_MS))
+
+        // Delay the appearance of the check icon
+        val checkAlpha: Float by animateFloatAsState(
+            targetValue = if (showSpinner) 0f else 1f,
+            animationSpec = tween(
+                delayMillis = CHECK_ALPHA_DURATION_MS,
+                durationMillis = CHECK_ALPHA_DURATION_MS,
+                easing = LinearEasing,
+            ),
+            label = "check_icon_alpha"
+        )
+
+        // Fade out loading spinner
+        AnimatedVisibility(
+            visible = showSpinner,
+            enter = enterTransition,
+            exit = exitTransition,
+        ) {
+            V3LoadingSpinner(
+                modifier = Modifier.size(56.dp)
+            )
+        }
+
+        // Fade in + slide success content.
+        AnimatedVisibility(
+            visible = !showSpinner,
+            enter = enterTransition + slideInVertically(initialOffsetY = { it / SLIDE_IN_ANIMATION_FRACTION }),
+            exit = exitTransition
+        ) {
+            SuccessCompletedContent(
+                checkAlpha = checkAlpha,
+                customSuccessMessage = customSuccessMessage,
+                accountsCount = accountsCount
+            )
+        }
+    }
+}
+
+@Composable
+private fun SuccessCompletedContent(
+    checkAlpha: Float,
```

Comment:
Can we avoid recomposition [by passing in the alpha](https://developer.android.com/jetpack/compose/animation/quick-guide#optimize-performance) as a lambda into `graphicsLayer { … }`?

Reply:
good catch - https://github.com/stripe/stripe-android/pull/7761/commits/75a7f0710ab6c02823ce18d18395b96bf0207e3f